drivers/ecg/adas1000/adas1000.c

```diff
@@ -0,0 +1,438 @@
+			    ADAS1000_16KHZ_FRAME_SIZE;
+		break;
+	case ADAS1000_128KHZ_FRAME_RATE:
+		*spi_freq = ADAS1000_128KHZ_FRAME_RATE *
+			    ADAS1000_128KHZ_WORD_SIZE *
+			    ADAS1000_128KHZ_FRAME_SIZE;
+		break;
+	case ADAS1000_31_25HZ_FRAME_RATE:
+		*spi_freq = (ADAS1000_31_25HZ_FRAME_RATE *
+			     ADAS1000_31_25HZ_WORD_SIZE *
+			     ADAS1000_31_25HZ_FRAME_SIZE) / 100;
+		break;
+	default: /** ADAS1000_2KHZ__FRAME_RATE */
+		*spi_freq = ADAS1000_2KHZ_FRAME_RATE *
+			    ADAS1000_2KHZ_WORD_SIZE *
+			    ADAS1000_2KHZ_FRAME_SIZE;
+		break;
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Initializes the SPI communication with ADAS1000. The ADAS1000
+ *	  is configured with the spefified frame rate and all the words in a
+ *	  frame are activated.
+ * @param device - the device structure.
+ * @param init_param - the initialization parameters.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t adas1000_init(struct adas1000_dev **device,
+		      const struct adas1000_init_param *init_param)
+{
+	int32_t ret;
+	struct adas1000_dev *dev;
+
+	dev = (struct adas1000_dev *)calloc(1, sizeof(*dev));
+	if (!dev)
+		return FAILURE;
+
+	/** store the selected frame rate */
+	dev->frame_rate = init_param->frame_rate;
+
+	/** Initialize the SPI controller. */
+	ret = spi_init(&dev->spi_desc, &init_param->spi_init);
+	if (ret != SUCCESS) {
+		free(dev);
+		return ret;
+	}
+
+	/** Reset the ADAS1000. */
+	ret = adas1000_soft_reset(dev);
+	if (ret != SUCCESS)
+		return ret;
+
+	/** Activate all the channels */
+	ret = adas1000_set_inactive_framewords(dev, ADAS1000_ALL_CH_MASK);
+	if (ret != SUCCESS)
+		return ret;
+
+	/** Set the frame rate */
+	ret = adas1000_set_frame_rate(dev, dev->frame_rate);
+	if (ret != SUCCESS)
+		return ret;
+
+	*device = dev;
+
+	return ret;
+}
+
+/**
+ * @brief Read device register.
+ * @param device - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The data read from the register.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t adas1000_read(struct adas1000_dev *device, uint8_t reg_addr,
+		      uint32_t *reg_data)
+{
+	int32_t ret;
+	uint8_t i;
+	uint8_t buff[4];
+	uint8_t buff_size = 4;
+
+	*reg_data = 0;
+
+	buff[0] = reg_addr;
+
+	ret = spi_write_and_read(device->spi_desc, buff, buff_size);
+	if(ret)
+		return FAILURE;
+
+	for(i = 1; i < buff_size; i++)
+		*reg_data = (*reg_data << 8) | buff[i];
+
+	return ret;
+}
+
+/**
+ * @brief Write device register.
+ * @param device - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The data to be written.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t adas1000_write(struct adas1000_dev *device, uint8_t reg_addr,
+		       uint32_t reg_data)
+{
+	uint8_t i;
+	uint8_t buff[4];
+	uint8_t buff_size = 4;
+
+	buff[0] = ADAS1000_COMM_WRITE | reg_addr;
+
+	for (i = 1; i < buff_size; i++)
```

Comment:
Nitcick: I think is easy to read and faster to set each element because the are only 3.
buff[1] = reg_data >> 16;
buff[2] = reg_data >> 8;
buff[3] = reg_data;

Reply:
For this particular case, yes. But I preferred using a generic approach for when the number of bytes is variable.